src/utils/specs.py

```diff
@@ -52,129 +53,264 @@
+    _rocminfo = rocminfo_full.split("\n")
+    gpu_arch, idx = detect_arch(_rocminfo)
+    _rocminfo = _rocminfo[idx + 1 :] # update rocminfo for target section
+    specs = MachineSpecs(version=specs_version,
+                        timestamp=timestamp,
+                        _rocminfo=_rocminfo,
+                        hostname=hostname,
+                        cpu_model=cpu_model,
+                        sbios=sbios,
+                        linux_kernel_version=linux_kernel_version,
+                        amd_gpu_kernel_version=amd_gpu_kernel_version,
+                        cpu_memory=cpu_memory,
+                        gpu_memory=gpu_memory,
+                        linux_distro=linux_distro,
+                        rocm_version=rocm_version,
+                        vbios=vbios,
+                        compute_partition=compute_partition,
+                        memory_partition=memory_partition,
+                        gpu_arch=gpu_arch)
+    # Load above SoC specs via module import
+    try:
+        soc_module = importlib.import_module('omniperf_soc.soc_'+ specs.gpu_arch)
+    except ModuleNotFoundError as e:
+        error("Arch %s marked as supported, but couldn't find class implementation %s." % (specs.gpu_arch, e))
+    soc_class = getattr(soc_module, specs.gpu_arch+'_soc')
+    soc_obj = soc_class(args, specs)
+    # Update arch specific specs
+    specs.total_l2_chan: str = total_l2_banks(
+        specs.gpu_model, int(specs._l2_banks), specs.memory_partition
+    )
+    specs.hbm_bw: str = str(int(specs.max_mclk) / 1000 * 32 * specs.get_hbm_channels())
+    return specs
+
+@dataclass(kw_only=True)
+class MachineSpecs:
+    ##########################################
+    ## A. Workload / Spec info
+    ##########################################
+
+    # these three fields are special in that they're not included
+    # when you use (e.g.,) --specs to view the machinespecs, but they
+    # _are_ included in profiling/analysis, so we mark them as 'optional'
+    # in the metadata to avoid erroring out on missing fields on
+    # serialization
+    workload_name: str = field(default=None, metadata={
+        'doc': 'The name of the workload data was collected for.',
+        'name': 'Workload Name',
+        'optional': True})
+    command: str = field(default=None, metadata={
+        'doc': 'The command the workload was executed with.',
+        'name': 'Command',
+        'optional': True})
+    ip_blocks: str = field(default=None, metadata={
+        'doc': 'The hardware blocks profiling information was collected for.',
+        'name': 'IP Blocks',
+        'optional': True
+        })
+    timestamp: str = field(default=None, metadata={
+        'doc': 'The time (in local system time) when data was collected',
+        'name': 'Timestamp'})
+    version: str = field(default=None, metadata={
+        'doc': 'The version of the machine specification file format.',
+        'name': 'MachineSpecs Version'})
```

Comment:
OCD, but I'm not crazy about embedding version directly in the table. Maybe just printing it above like
![image](https://github.com/AMDResearch/omniperf/assets/11466906/f53e4371-ee29-4267-ac98-3e501870837b)

I'll see if that can be changed

Reply:
Ah, yeah -- that's doable. I can push a patch, 1 sec

Reply:
![image](https://github.com/AMDResearch/omniperf/assets/6463881/273a729c-063a-4a50-879e-3210eba05118)

Reply:
Done in 0b566151dade65b63857467a93e661047342facc